From f3c13c23bf09bc49bc52b0ba9609d5b6a2833c0a Mon Sep 17 00:00:00 2001
From: Matthew Waters <matthew@centricular.com>
Date: Mon, 8 Oct 2018 17:18:56 +1100
Subject: [PATCH] meson: add option for disabling GIO sniffing

Some linux-like platforms (e.g. Android) don't have a mime-info database
readily available for use.
---
 meson.build       | 8 ++++----
 meson_options.txt | 5 +++++
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/meson.build b/meson.build
index 43b11f6..6b0ba49 100644
--- a/meson.build
+++ b/meson.build
@@ -189,11 +189,11 @@ gmodule_dep = dependency('gmodule-no-export-2.0')
 gio_dep = dependency('gio-2.0')
 
 # On non-Windows/macOS systems we always required shared-mime-info and GIO
-if host_system != 'windows' and host_system != 'darwin'
-  shared_mime_dep = dependency('shared-mime-info')
+# shared_mime_dep = []
+shared_mime_dep = []
+if get_option('gio_sniffing') and host_system != 'windows' and host_system != 'darwin'
+  shared_mime_dep += dependency('shared-mime-info')
   gdk_pixbuf_conf.set('GDK_PIXBUF_USE_GIO_MIME', 1)
-else
-  shared_mime_dep = []
 endif
 
 # Check if medialib is available
diff --git a/meson_options.txt b/meson_options.txt
index 9f6b650..a7c33a8 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -46,3 +46,8 @@ option('installed_tests',
        description: 'Install the test suite',
        type: 'boolean',
        value: true)
+option('gio_sniffing',
+       description: 'Perform file type detection using GIO (Unused on MacOS and Windows)',
+       type: 'boolean',
+       value: true)
+
-- 
2.19.0

